src/vmm/src/rpc_interface.rs

```diff
@@ -325,6 +319,26 @@ impl<'a> PrebootApiController<'a> {
             CreateSnapshot(_) | SendCtrlAltDel => Err(VmmActionError::OperationNotSupportedPreBoot),
         }
     }
+
+    #[cfg(target_arch = "x86_64")]
+    fn load_snapshot(&mut self, load_params: LoadSnapshotParams) -> ActionResult {
+        let load_start_us = utils::time::get_time_us(utils::time::ClockType::Monotonic);
```

Comment:
load_start_us should be computed on the API thread. In case of high CPU load we expect some latency here and we should see it in the metrics.

Reply:
> load_start_us should be computed on the API thread

I disagree:
- That would break modularity and spread the logic over multiple components. It would also add vmm-specific logic to the otherwise completely decoupled api server.
- The metric itself will lose granularity in regards to the snapshot operation. This metric should ideally measure exclusively snapshot code.
- Regressions or inadequate performance in the API server code or the API<->VMM communication should have their own metrics.

Reply:
I don't fully disagree with @acatangiu's point, but from an UX point of view I want to be able to get see how much time does it take since receiving the API call up until the snapshot is loaded. 

I do not have a hard preference on how this should be implemented, if it is possible to make them modular so one can  add api_latency_us + load_time_us + resume_time_us and get the e2e timing.